src/ui/win32/bindings/WindowBinding.cpp

```diff
@@ -0,0 +1,237 @@
+                break;
+
+            case RelativePosition::Before:
+                oPosition.Y = rcMainWindow.top - oSize.Height - 4;
+                break;
+
+            case RelativePosition::Near:
+                oPosition.Y = rcMainWindow.top;
+                break;
+
+            default:
+            case RelativePosition::Center:
+                oPosition.Y = (rcMainWindow.bottom + rcMainWindow.top - oSize.Height) / 2;
+                break;
+
+            case RelativePosition::Far:
+                oPosition.Y = rcMainWindow.bottom - oSize.Height;
+                break;
+
+            case RelativePosition::After:
+                oPosition.Y = rcMainWindow.bottom;
+                break;
+        }
+    }
+
+    // make sure the window is on screen
+    auto offset = oPosition.X + oSize.Width - (oWorkAreaPosition.X + oWorkAreaSize.Width);
+    if (offset > 0)
+        oPosition.X -= offset;
+
+    offset = oWorkAreaPosition.X - oPosition.X;
+    if (offset > 0)
+        oPosition.X += offset;
+
+    offset = oPosition.Y + oSize.Height - (oWorkAreaPosition.Y + oWorkAreaSize.Height);
+    if (offset > 0)
+        oPosition.Y -= offset;
+
+    offset = oWorkAreaPosition.Y - oPosition.Y;
+    if (offset > 0)
+        oPosition.Y += offset;
+
+    // temporarily clear out the key while we reposition so OnSizeChanged and OnPositionChanged ignore the message
+    std::string sSizeAndPositionKey = m_sSizeAndPositionKey;
+    m_sSizeAndPositionKey.clear();
+
+    SetWindowPos(m_hWnd, nullptr, oPosition.X, oPosition.Y, oSize.Width, oSize.Height, 0);
+
+    m_sSizeAndPositionKey = sSizeAndPositionKey;
+}
+
+void WindowBinding::OnSizeChanged(_UNUSED ra::ui::Size oSize)
+{
+    if (!m_sSizeAndPositionKey.empty())
+    {
+        // oSize is the size of client area, we need the size of the window (including non-client area)
+        RECT rcDialog;
+        GetWindowRect(m_hWnd, &rcDialog);
+        ra::ui::Size oDialogSize{ rcDialog.right - rcDialog.left, rcDialog.bottom - rcDialog.top };
+
+        ra::services::ServiceLocator::GetMutable<ra::services::IConfiguration>().SetWindowSize(m_sSizeAndPositionKey, oDialogSize);
+    }
+}
+
+void WindowBinding::OnPositionChanged(_UNUSED ra::ui::Position oPosition)
+{
+    if (!m_sSizeAndPositionKey.empty())
+    {
+        // oPosition is topleft corner of client area, we need topleft corner of window (including non-client area)
+        RECT rcDialog;
+        GetWindowRect(m_hWnd, &rcDialog);
+
+        // capture position relative to main window
+        RECT rcMainWindow;
+        GetWindowRect(g_RAMainWnd, &rcMainWindow);
+        ra::ui::Position oRelativePosition{ rcDialog.left - rcMainWindow.left, rcDialog.top - rcMainWindow.top };
+
+        ra::services::ServiceLocator::GetMutable<ra::services::IConfiguration>().SetWindowPosition(m_sSizeAndPositionKey, oRelativePosition);
+    }
+}
+
+void WindowBinding::OnViewModelStringValueChanged(const StringModelProperty::ChangeArgs& args)
```

Comment:
this is the `NotifyTarget` callback. The first statement handles the window title, then the labels are managed by an unordered_map.